eng/build.cmd

```diff
@@ -0,0 +1,2 @@
+@echo off
+powershell -ExecutionPolicy ByPass -NoProfile -command "& """%~dp0build.ps1""" %*"
```

Comment:
> Why adding this one when we also have the root one?

The root `build.cmd` command unconditionally passes the `-build` argument which I explicitly do not want here. 

> Also can we just get rid of all the .cmd files since the .ps1 files work just as well?

The reason for CMD files here is just following the existing pattern. The last time I tried powershell files was so long ago I've forgotten the reasons I went away from it. If the task works well now I'm happy to switch to them. Prefer to do that in a separate PR though cause I woudl want to flip the official pipeline over as well and that requires a different type of hair pulling 😄

Reply:
For tracking https://github.com/dotnet/roslyn/issues/48117